pkg/discovery/resolvermgr/grpc_resolver_mgr.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolvermgr
+
+import (
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/naming"
```

Comment:
btw, `naming.Resolver` is deprecated: https://github.com/grpc/grpc-go/blob/master/naming/naming.go#L53

I think we should upgrade to grpc 1.20 before continuing this work

Reply:
I can update the code first to use resolver package first and then we could upgrade grpc version, what do you think?

Reply:
in that case I would suggest removing resolver implementation from this PR and just merging the builder changes that introduce a resolver, because making discovery work with old API doesn't make sense.